st3/lsp_utils/helpers.py

```diff
@@ -93,3 +95,47 @@ def log_and_show_message(message: str, additional_logs: Optional[str] = None, sh
     print(message, '\n', additional_logs) if additional_logs else print(message)
     if show_in_status:
         sublime.active_window().status_message(message)
+
+
+def unique_everseen(iterable: Iterable[_T], key: Optional[Callable[[_T], Any]] = None) -> Generator[_T, None, None]:
```

Comment:
This is taken from https://github.com/more-itertools/more-itertools/blob/7e46c39ea5d73b80f36ef2faf266e10108029b02/more_itertools/recipes.py#L478